include/wx/private/uilocale.h

```diff
@@ -23,6 +23,11 @@ const wxLanguageInfos& wxGetLanguageInfos();
 // Function returning hard-coded values for the "C" locale.
 wxString wxGetStdCLocaleInfo(wxLocaleInfo index, wxLocaleCategory cat);
 
+#ifdef __WXOSX_COCOA__
+// Function returning a pointer to the current NSLocale
+WXDLLIMPEXP_BASE void* wxGetCurrentNSLocale();
+#endif
+
```

Comment:
I don't think this belongs to this header, it is Mac-, rather than locale-, specific.

Putting it in a (new) header under `include/wx/osx/private` would result in better physical (de)coupling and would allow to use `NSLocale*` as the function return type and so avoid the ugly casts too.

Reply:
> I don't think this belongs to this header, it is Mac-, rather than locale-, specific.
> 
> Putting it in a (new) header under `include/wx/osx/private` would result in better physical (de)coupling and would allow to use `NSLocale*` as the function return type and so avoid the ugly casts too.

Ok, this can be done, of course.